apps/studio/src/features/editing-experience/components/ComplexEditorStateDrawer.tsx

```diff
@@ -104,6 +112,69 @@ export default function ComplexEditorStateDrawer(): JSX.Element {
     setPreviewPageState(newPageState)
   }
 
+  const handleSave = async () => {
+    let newPageState = previewPageState
+
+    if (modifiedAssets.length > 0) {
+      const updatedBlocks = Array.from(previewPageState.content)
+      const newBlock = _.cloneDeep(updatedBlocks[currActiveIdx])
+
+      if (!newBlock) {
+        return
+      }
+
+      // Upload all new/modified images/files
+      const assetsToUpload = modifiedAssets.filter((asset) => !!asset.file)
+      try {
+        await Promise.all(
+          assetsToUpload.map(({ path, file }) => {
+            if (!file) {
+              return
+            }
+
+            return uploadAsset({ file }).then((res) =>
+              _.set(newBlock, path, res.path),
+            )
+          }),
+        )
+      } catch (e) {
+        // All the promises are rejected if any one of them fails
```

Comment:
Why should all the promises be rejected? Should we be using allSettled to upload as many as we can?

Reply:
Adjusted in 8752fd749c5b7b072e30d76e12b1e3c77dfa1943!